src/daft-core/src/series/ops/is_in.rs

```diff
@@ -0,0 +1,142 @@
+use std::collections::HashSet;
+
+use common_error::{DaftError, DaftResult};
+
+use crate::{datatypes::BooleanArray, IntoSeries, Series};
+
+macro_rules! do_is_in {
+    ($name:expr, $data:expr, $items:expr) => {{
+        let mut set = HashSet::new();
+        for i in 0..$items.len() {
+            set.insert($items.get(i));
+        }
+
+        let mut result = Vec::new();
+        for i in 0..$data.len() {
+            result.push(set.contains(&$data.get(i)));
+        }
+
+        Ok(BooleanArray::from(($name, result.as_slice())).into_series())
+    }};
+}
+
+impl Series {
+    pub fn is_in(&self, items: &Self) -> DaftResult<Series> {
+        let items = &items.list()?.flat_child;
+        let default =
+            BooleanArray::from((self.name(), vec![false; self.len()].as_slice())).into_series();
+        if items.is_empty() {
+            return Ok(default);
+        }
+        // match items.data_type() with the datatypes of LiteralValue because items is a List(Vec<LiteralValue>),
+        // attept to cast self to the same datatype as items, then check if self is in items
+        match items.data_type() {
```

Comment:
Should this not be the other way around?

1. Matching on `self.data_type()`
2. Attempt to coerce the type of `items` into `self.data_type()` for comparison

This is because the data in `self` is probably always much "better typed", and the data in `items` is passed in by the user via the Python `.is_in(...)` API and will need to be coerced.

Reply:
yep makes sense, i changed my logic to this: https://github.com/Eventual-Inc/Daft/pull/1811/files#diff-2330210aef0d266cd0fdcb54102535bd0bdcf08337fe0da49758239a28730532R18-R63, which follows the same typecasting logic that binary ops use: https://github.com/Eventual-Inc/Daft/blob/main/src/daft-core/src/series/array_impl/binary_ops.rs#L115